tools/submission/submission-checker.py

```diff
@@ -982,28 +987,29 @@ def check_power_dir(power_path, ranging_path, testing_path, scenario_fixed, more
-            # TODO: Currently, LoadGen does NOT print out the actual number of queries in detail logs. There is a
-            # "generated_query_count", but LoadGen exits early when the min_duration has been met, so it is not equal to
-            # the actual number of queries. For now, we will make use of "result_qps_with_loadgen_overhead", which is
-            # defined as: (sample_count - 1) / pr.final_query_issued_time, where final_query_issued_time can be
-            # approximated by power_duration (off by one query worth of latency, which is in general negligible compared
-            # to 600-sec total runtime and can be offsetted by removing the "+1" when reconstructing the sample_count).
-            # Not an ideal approach, but probably the best we can do in v1.0.
-            # As for MultiStream, it always runs for 270336 queries, so using "generated_query_count" as above is fine.
-            # In v1.1, we should make LoadGen print out the actual number of queries and use it to compute the J/sample 
-            # for SingleStream and MultiStream power metrics.
-            power_duration = (power_end - power_begin).total_seconds()
-            num_samples = mlperf_log["result_qps_with_loadgen_overhead"] * power_duration
-            power_metric = avg_power * power_duration / num_samples
+        else:
+            # In SingleStream and MultiStream scenarios, the power metric is in J/query.
+            assert scenario_fixed in ["MultiStream", "SingleStream"], "Unknown scenario: {:}".format(scenario_fixed)
+            if not config.has_query_count_in_log():
+                # Before v2.0, LoadGen does NOT print out the actual number of queries in detail logs. There is a
+                # "generated_query_count", but LoadGen exits early when the min_duration has been met, so it is not equal to
+                # the actual number of queries. To work around it, make use of "result_qps_with_loadgen_overhead", which is
+                # defined as: (sample_count - 1) / pr.final_query_issued_time, where final_query_issued_time can be
+                # approximated by power_duration (off by one query worth of latency, which is in general negligible compared
+                # to 600-sec total runtime and can be offsetted by removing the "+1" when reconstructing the sample_count).
+                # As for MultiStream, it always runs for 270336 queries, so using "generated_query_count" as above is fine.
+                if scenario_fixed in ["MultiStream"]:
+                    num_queries = mlperf_log["generated_query_count"] * mlperf_log["generated_samples_per_query"]
+                elif scenario_fixed in ["SingleStream"]:
+                    num_queries = mlperf_log["result_qps_with_loadgen_overhead"] * power_duration
+            else:
+                # Starting from v2.0, LoadGen logs the actual number of issued queries.
+                num_queries = mlperf_log["result_query_count"]
```

Comment:
This line uses the new feature @pgmpablo157321 added recently.

Let me run some local tests to make sure it works before we merge this.

Reply:
@nvpohanh Any conclusion yet?